Dockerfile

```diff
@@ -1,4 +1,6 @@
-FROM golang:1.21 as builder
+ARG NANOSERVER_VERSION=ltcs2022
```

Comment:
why only have the nano server version as the arg instead of the full image name, e.g. `ARG WINBASE=scratch` and then users have the ability to pass a different windows base image if they wanted instead of hardcoding to mcr/nanoserver